チャット画面ビュー作成

iwa77 opened the pull request:
# WHAT　
chat-spaceチャット画面のマークアップ
# WHY　
アプリのチャット画面実装のため

Commit:
first commit

iwa77 committed:
create-controller

iwa77 committed:
file-create

iwa77 committed:
scss-create

iwa77 committed:
font-awesome-install

iwa77 commented:
まだチャット画面実装できていない途中段階ですが　このタイミングでレビューをお願いしてもよいのでしょうか?

kenchan0503 commented on app/assets/stylesheets/_reset.scss:
```diff
@@ -0,0 +1,126 @@
+    TODO with the prefix, should I group by selector or property for weight savings?
+*/
+html{
+    color:#000;
```
複数箇所で使用している値は変数にまとめてあげましょう

iwa77 replied:
修正致しました

kenchan0503 commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,5 @@
+
+/*
+ *= require font-awesome
+ */
```
改行しましょう

kenchan0503 commented:
WIPでレビューを依頼する際は何を見てほしいかを明確にしましょう。
レビュワーも何をチェックして何をスルーすればいいのかわからないので。

iwa77 committed:
chat-view-revise

iwa77 committed:
haml-revise-for-server

localhost:3000に接続できなかったのと、hamlの文法エラーで正しく画面表示できなかったため、原因特定のためindex.htmlを、helloworldと画面表示されるようにした上で大幅にやり直し作業を加えた

iwa77 committed:
haml-revise

ハッシュロケット記法を直す上での修正

iwa77 committed:
view-making

iwa77 committed:
view-making-revise1

iwa77 commented:
![image](https://cloud.githubusercontent.com/assets/27888253/25576217/377ba1f6-2e98-11e7-8416-dc6dcdbbe7a2.png)

vip-take commented on Gemfile:
```diff
@@ -1,52 +1,26 @@
   gem "haml-rails"
   gem 'erb2haml'
-
+  gem "font-awesome-rails"
```
これはproductionでも使いますので、groupから出して、上に持って行きましょう！

iwa77 replied:
了解致しました！　修正しました！

vip-take commented on app/assets/stylesheets/_top-content.scss:
```diff
@@ -0,0 +1,7 @@
+  width: 100%;
+  position: fixed;
+  text-align: center;
+}
```
改行しましょー！

iwa77 replied:
ご指摘ありがとうございます！修正いたしました！

vip-take commented on app/assets/stylesheets/modules/_chat-content.scss:
```diff
@@ -0,0 +1,44 @@
+$base-color: #ffffff; 
```
変数は１つのscssファイルにまとめてあげましょう！
その方が管理しやすいです。詳しくは、カリキュラムの「Sassを学習しよう」をご確認ください！

iwa77 replied:
変数まとめてみました！

vip-take commented on app/assets/stylesheets/modules/_chat-content.scss:
```diff
@@ -0,0 +1,44 @@
+      
+    }
+
+    &-icon1{
```
あと参考までにですが、こんな要素の指定の仕方もあったりします！
http://qiita.com/ituki_b/items/62a752389385de7ba4a2

vip-take replied:
あとあまり、数字入れるのは良くないですね。

vip-take commented on app/assets/stylesheets/modules/_top-content.scss:
```diff
@@ -0,0 +1,10 @@
+    color: #ffffff;
+    text-align: center;
+  }
+  
```
改行を入れましょう！

vip-take commented on app/assets/stylesheets/modules/_main-content.scss:
```diff
@@ -0,0 +1,85 @@
+    }
+  }
+}
+  
```
こちらも改行を！

vip-take commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,45 @@
+.top-content
+  %p チャットグループが作成されました
+.chat-content
```
左カラムに当たる部分は、部分テンプレート化してあげましょう！

iwa77 replied:
修正致しました！

iwa77 committed:
view-making-revise-2

ta910 commented on app/assets/stylesheets/modules/_chat-content.scss:
```diff
@@ -0,0 +1,44 @@
+@import "./modules/chat-variable";
```
この一行はapplication.scssの中で、これを記述したファイルより上で読み込んでおけば必要ありません。

iwa77 replied:
修正致しました！

ta910 commented on app/assets/stylesheets/modules/_main-content.scss:
```diff
@@ -0,0 +1,81 @@
+  float: right;
+  height:100vh;
+
+  .chat-main__header{
```
BEM記法を用いましょう！

iwa77 replied:
修正致しました！

ta910 commented on app/assets/stylesheets/modules/_top-content.scss:
```diff
@@ -0,0 +1,20 @@
+@import "./modules/chat-variable";
```
こちらもですね。

iwa77 replied:
この場合　top-content　が先に読まれますでしょうか？そう思って　main-contentのほうをなくしました！

seoky0 replied:
importはapplication.scssで全て読み込んでしまいましょう

ta910 commented on app/assets/stylesheets/modules/_top-content.scss:
```diff
@@ -0,0 +1,20 @@
+
+}
+
+
```
余分な空行はPRを出す前に消しておきましょう。

iwa77 replied:
修正致しました！

ta910 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,43 @@
+    %i.fa.fa-picture-o
+    %input.send-btn{ type: :submit, value: "send" }
+   
+    
```
ここらへんも余分ですね。

iwa77 replied:
修正致しました！

ta910 commented on config/environments/development.rb:
```diff
@@ -51,4 +51,8 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+ 
```
ここもです！

iwa77 replied:
修正致しました！

iwa77 committed:
view-making-revise-3

iwa77 committed:
view-making-revise-4

yuya1212h4 commented:
チャット画面について、全体を見えるようにキャプチャをもう一度貼って頂くようにお願い致します。

iwa77 commented:
![image](https://cloud.githubusercontent.com/assets/27888253/25579115/c568aa24-2eaf-11e7-922c-8db8eec25de9.png)

yuya1212h4 commented on app/assets/stylesheets/modules/_chat-content.scss:
```diff
@@ -0,0 +1,42 @@
+    color: $base-color;
+
+    .group-name{
+      margin:20px 20px 5px 20px;
```
こちらのコロンの後のスペースを入れて頂いて、全体として統一して頂けるようにお願い致します。

iwa77 replied:
修正致しました！

yuya1212h4 commented on app/assets/stylesheets/modules/_chat-content.scss:
```diff
@@ -0,0 +1,42 @@
+    }
+
+  }
+
```
この空行についても先程と同様に修正をお願い致します。

iwa77 replied:
修正致しました！

iwa77 commented:
大変失礼しました！貼ってみました！

yuya1212h4 commented on app/assets/stylesheets/modules/_top-content.scss:
```diff
@@ -0,0 +1,19 @@
+    color: $base-color;
+    text-align: center;
+  }
+
```
この部分の修正を忘れずにお願い致します。

iwa77 replied:
修正致しました！

yuya1212h4 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,40 @@
+
+
+
+
```
こちらの余計な空行についても削除の方をよろしくお願い致します。

iwa77 replied:
失礼いたしました　削除しました！

iwa77 committed:
view-making-revise-5

iwa77 committed:
view-making-revise6

パディングの値が間違っていると気づき修正致しましたすみません

iwa77 committed:
view-making-revise7

seoky0 commented on app/assets/stylesheets/modules/_chat-variable.scss:
```diff
@@ -0,0 +1,2 @@
+$base-color: #ffffff; 
+$accent-color: #38aef0;
```
一番下には空行をいれましょう

iwa77 replied:
修正致しました！

seoky0 commented on app/assets/stylesheets/modules/_main-content.scss:
```diff
@@ -0,0 +1,79 @@
+      font-size: 16px;
+      display: inline-block; 
+      margin-right: 10px;  
+      color: #434a54; 
```
変数化しましょう

iwa77 replied:
修正致しました！

seoky0 commented on app/controllers/messages_controller.rb:
```diff
@@ -0,0 +1,2 @@
+class MessagesController < ApplicationController
```
アクションを書いてあげましょう

iwa77 replied:
申し訳ありません　修正致しました！

seoky0 commented on app/views/layouts/_top-content.html.haml:
```diff
@@ -0,0 +1,2 @@
+.top-content
+  %p チャットグループが作成されました
```
下に空行いれましょう

iwa77 replied:
修正致しました！

seoky0 commented:
以前貼っていただいたキャプチャだと、ボタンに枠がついているのですが、最新の状態では消せていますか？
消せていない場合はこちらを修正しましょう。

iwa77 commented:
SENDボタンに枠があったため、枠を消して修正致しました！

iwa77 committed:
view-maikng-revise8

miyagit commented:
現在のビュー画面はフラッシュメッセージ込みのビュー画面を作成していますでしょうか(headerの青背景のグループが作成されましたのことを指しています。)？

iwa77 commented:
いえ！　top-contentでしょうか？　チャットグループが作成されました　がずっと表示されてる状態です！

seoky0 commented:
承知しました！
LGTMです！
